source/Mission.cpp

```diff
@@ -1436,8 +1438,16 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 			+ Identifier() + "\" uses invalid " + std::move(reason));
 		return result;
 	}
+
+	int64_t escortShips = 0;
 	for(const NPC &npc : npcs)
-		result.npcs.push_back(npc.Instantiate(subs, sourceSystem, result.destination->GetSystem(), jumps, payload));
+	{
+		result.npcs.push_back(npc.Instantiate(subs, sourceSystem, result.destination->GetSystem(), jumps,
+			payload, escortPayload));
```

Comment:
Is it intentional that the NPCs each get a different payload?

Reply:
We don't update the `payload` until after this loop. Every NPC gets the same payload and escortPaylaod input, then any NPC actions will account for the escorts in that NPC block, then we update the `payload` for the rest of the actions in the mission to account for the escorts in every NPC block. This is so that we don't need to split NPC instantiation between just instantiating the ships and then instantiating everything else about the NPC.

Reply:
Oh yeah, I meant like the final `payload` includes every NPC ships, but the individual `payload` calculated in NPC only includes that NPC's ships and not the rest (if that makes sense). If that's intentional then all good just wanted to double check ^^